File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/spendingsections/addeditspendingsection/AddEditSpendingSectionContract.java
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.spendingsections.addeditspendingsection;

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter;
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView;
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SpendingSection;

/**
 * This specifies the contract between the view and the homePresenter.
 */
public interface AddEditSpendingSectionContract {
    interface View extends BaseView<Presenter> {

        void showAddSuccess();

        void showEditSuccess();
    }

    interface Presenter extends BasePresenter<View> {
        void addSpendingSection(SpendingSection spendingSection);

        void editSpendingSection(Integer id, SpendingSection spendingSection);
    }
}

File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/spendingsections/addeditspendingsection/AddEditSpendingSectionContract.kt
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.spendingsections.addeditspendingsection

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SpendingSection

/**
 * This specifies the contract between the view and the homePresenter.
 */
interface AddEditSpendingSectionContract {
    interface View : BaseView<Presenter> {

        fun showAddSuccess()

        fun showEditSuccess()
    }

    interface Presenter : BasePresenter<View> {
        fun addSpendingSection(spendingSection: SpendingSection)

        fun editSpendingSection(id: Int?, spendingSection: SpendingSection)
    }
}
